[2024-09-09T16:45:11.208+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_job_dag.run_spark_job manual__2024-09-09T16:45:10.516652+00:00 [queued]>
[2024-09-09T16:45:11.221+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_job_dag.run_spark_job manual__2024-09-09T16:45:10.516652+00:00 [queued]>
[2024-09-09T16:45:11.221+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 2
[2024-09-09T16:45:11.246+0000] {taskinstance.py:1382} INFO - Executing <Task(SparkSubmitOperator): run_spark_job> on 2024-09-09 16:45:10.516652+00:00
[2024-09-09T16:45:11.253+0000] {standard_task_runner.py:57} INFO - Started process 275 to run task
[2024-09-09T16:45:11.258+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'spark_job_dag', 'run_spark_job', 'manual__2024-09-09T16:45:10.516652+00:00', '--job-id', '9', '--raw', '--subdir', 'DAGS_FOLDER/simples3bucketpipeline.py', '--cfg-path', '/tmp/tmp7r_vee57']
[2024-09-09T16:45:11.263+0000] {standard_task_runner.py:85} INFO - Job 9: Subtask run_spark_job
[2024-09-09T16:45:11.345+0000] {task_command.py:416} INFO - Running <TaskInstance: spark_job_dag.run_spark_job manual__2024-09-09T16:45:10.516652+00:00 [running]> on host 7cde20e91372
[2024-09-09T16:45:11.438+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spark_job_dag' AIRFLOW_CTX_TASK_ID='run_spark_job' AIRFLOW_CTX_EXECUTION_DATE='2024-09-09T16:45:10.516652+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-09-09T16:45:10.516652+00:00'
[2024-09-09T16:45:11.448+0000] {spark_submit.py:273} INFO - Could not load connection string spark_default, defaulting to yarn
[2024-09-09T16:45:11.449+0000] {spark_submit.py:403} INFO - Spark-Submit cmd: spark-submit --master yarn --conf spark.master=spark://spark-master:7077 --executor-cores 2 --executor-memory 2g --name spark_job --verbose /tmp/spark_job.py
[2024-09-09T16:45:11.590+0000] {spark_submit.py:579} INFO - /home/***/.local/lib/python3.8/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2024-09-09T16:45:11.591+0000] {spark_submit.py:579} INFO - JAVA_HOME is not set
[2024-09-09T16:45:11.603+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 174, in execute
    self._hook.submit(self.application)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 505, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --conf spark.master=spark://spark-master:7077 --executor-cores 2 --executor-memory 2g --name spark_job --verbose /tmp/spark_job.py. Error code is: 1.
[2024-09-09T16:45:11.607+0000] {taskinstance.py:1400} INFO - Marking task as UP_FOR_RETRY. dag_id=spark_job_dag, task_id=run_spark_job, execution_date=20240909T164510, start_date=20240909T164511, end_date=20240909T164511
[2024-09-09T16:45:11.617+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 9 for task run_spark_job (Cannot execute: spark-submit --master yarn --conf spark.master=spark://spark-master:7077 --executor-cores 2 --executor-memory 2g --name spark_job --verbose /tmp/spark_job.py. Error code is: 1.; 275)
[2024-09-09T16:45:11.647+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2024-09-09T16:45:11.659+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
